src/NuGet.Core/NuGet.Packaging/Signing/SigningUtility.cs

```diff
@@ -71,30 +66,31 @@ public static bool IsCertificatePublicKeyValid(X509Certificate2 certificate)
                 throw new ArgumentNullException(nameof(extraStore));
             }
 
+            X509ChainStatus[] chainStatusList;
             using (var chain = new X509Chain())
             {
                 SetCertBuildChainPolicy(
-                    chain,
+                    chain.ChainPolicy,
                     extraStore,
                     DateTime.Now,
                     NuGetVerificationCertificateType.Signature);
 
-                if (chain.Build(certificate))
+                if (SigningUtility.BuildCertificateChain(chain, certificate, out chainStatusList))
                 {
                     return GetCertificateChain(chain);
                 }
+            }
 
-                foreach (var chainStatus in chain.ChainStatus)
+            foreach (var chainStatus in chainStatusList)
+            {
+                if (chainStatus.Status != X509ChainStatusFlags.NoError)
                 {
-                    if (chainStatus.Status != X509ChainStatusFlags.NoError)
-                    {
-                        throw new SignatureException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorInvalidCertificateChain, chainStatus.Status.ToString()));
-                    }
+                    throw new SignatureException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorInvalidCertificateChain, chainStatus.Status.ToString()));
```

Comment:
I thought you were going to use `X509ChainStatus.StatusInformation` for displayed errors and `X509ChainStatus.Status` for verbose logging.  Wouldn't this bubble up the enum member name in the error message?

Reply:
This method is not used in the verification code path. It is used in sign which I will address in a later PR. I just updated this method to account for the changes I made to the chain building, which were breaking it.